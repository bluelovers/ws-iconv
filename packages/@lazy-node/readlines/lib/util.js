"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.splitBufferByBuffer = exports.bufferIndexWith = exports.bufferIndexWithByBuffer = exports.bufferStripEndWith = exports.bufferStripEndWithByBuffer = exports.bufferEndWith = exports.bufferEndWithByBuffer = exports.handleOptionNewLineCharacter = void 0;
function handleOptionNewLineCharacter(newLineCharacter) {
    if (!newLineCharacter) {
        return [10 /* EnumNewLineCharacter.LF */];
    }
    else if (typeof newLineCharacter !== 'number') {
        let ls = [];
        for (let i = 0; i < newLineCharacter.length; i++) {
            let c = newLineCharacter[i];
            if (typeof c !== 'number') {
                ls.push(c.charCodeAt(0));
            }
            else {
                ls.push(c);
            }
        }
        return ls;
    }
    return [newLineCharacter];
}
exports.handleOptionNewLineCharacter = handleOptionNewLineCharacter;
function bufferEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWithByBuffer = bufferEndWithByBuffer;
function bufferEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[buf.length - 1] === value;
    }
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWith = bufferEndWith;
function bufferStripEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    if (i !== -1 && (i + value.length) === buf.length) {
        return buf.slice(0, i);
    }
    return buf;
}
exports.bufferStripEndWithByBuffer = bufferStripEndWithByBuffer;
function bufferStripEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        if (buf[buf.length - 1] === value) {
            return buf.slice(0, buf.length - 1);
        }
    }
    else {
        let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
        if (i !== -1 && (i + value.length) === buf.length) {
            /*
            console.dir({
                i,
                vl: value.length,
                bl: buf.length,

                v: value.toString(),
                b: buf.toString(),
                b2: buf.slice(0, i).toString(),
            })
             */
            return buf.slice(0, i);
        }
    }
    return buf;
}
exports.bufferStripEndWith = bufferStripEndWith;
function bufferIndexWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWithByBuffer = bufferIndexWithByBuffer;
function bufferIndexWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[byteOffset] === value;
    }
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWith = bufferIndexWith;
function splitBufferByBuffer(buffer, value) {
    const lines = [];
    let bufferPosition = 0;
    let lastNewLineBufferPosition = 0;
    while (true) {
        let line;
        let bufferPositionValue = buffer[bufferPosition];
        if (bufferIndexWithByBuffer(buffer, value, bufferPosition++)) {
            line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
            lines.push(line);
            lastNewLineBufferPosition = bufferPosition;
        }
        else if (bufferPositionValue === undefined) {
            break;
        }
    }
    let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
    if (leftovers.length) {
        lines.push(leftovers);
    }
    return lines;
}
exports.splitBufferByBuffer = splitBufferByBuffer;
//# sourceMappingURL=util.js.map