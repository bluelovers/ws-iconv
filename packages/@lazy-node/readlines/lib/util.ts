import { INewLineCharacter, EnumNewLineCharacter } from './types';

export function handleOptionNewLineCharacter(newLineCharacter?: INewLineCharacter): number[]
{
	if (!newLineCharacter)
	{
		return [EnumNewLineCharacter.LF];
	}
	else if (typeof newLineCharacter !== 'number')
	{
		let ls = [] as number[]

		for (let i = 0; i < newLineCharacter.length; i++)
		{
			let c = newLineCharacter[i];

			if (typeof c !== 'number')
			{
				ls.push(c.charCodeAt(0))
			}
			else
			{
				ls.push(c)
			}
		}

		return ls
	}

	return [newLineCharacter]
}

export function bufferEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		return buf[buf.length - 1] === value
	}

	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferStripEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	if (i !== -1 && (i + value.length) === buf.length)
	{
		return buf.slice(0, i)
	}

	return buf
}

export function bufferStripEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		if (buf[buf.length - 1] === value)
		{
			return buf.slice(0, buf.length - 1);
		}
	}
	else
	{
		let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

		if (i !== -1 && (i + value.length) === buf.length)
		{
			/*
			console.dir({
				i,
				vl: value.length,
				bl: buf.length,

				v: value.toString(),
				b: buf.toString(),
				b2: buf.slice(0, i).toString(),
			})
			 */

			return buf.slice(0, i)
		}
	}

	return buf
}

export function bufferIndexWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function bufferIndexWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	if (typeof value === 'number')
	{
		return buf[byteOffset] === value
	}

	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function splitBufferByBuffer(buffer: Buffer, value: Uint8Array)
{
	const lines: Buffer[] = [];
	let bufferPosition = 0;

	let lastNewLineBufferPosition = 0;
	while (true)
	{
		let line: Buffer;
		let bufferPositionValue = buffer[bufferPosition];

		if (bufferIndexWithByBuffer(buffer, value, bufferPosition++))
		{
			line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
			lines.push(line);
			lastNewLineBufferPosition = bufferPosition;
		}
		else if (bufferPositionValue === undefined)
		{
			break;
		}
	}

	let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
	if (leftovers.length)
	{
		lines.push(leftovers);
	}

	return lines;
}
