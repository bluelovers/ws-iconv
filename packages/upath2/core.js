"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const _path = require("path");
const type_1 = require("./lib/type");
exports.types = type_1.default;
class PathWrap {
    constructor(path, id) {
        this.sep = '/';
        let _static = getStatic(this);
        this[type_1.ORIGIN_KEY] = path;
        this.name = id;
        delete this[id];
        this[id] = this;
        Object.defineProperty(this, type_1.ORIGIN_KEY, { enumerable: false, });
        this.fn = Object.assign(this.__proto__, path, _static.fn);
    }
    join(path, ...paths) {
        return _this_origin(this).join(path, ...paths).replace(/\\/g, this.sep);
    }
    normalize(path) {
        return _this_origin(this).normalize(path).replace(/\\/g, this.sep);
    }
    relative(from, to) {
        return _this_origin(this).relative(from.toString(), to.toString()).replace(/\\/g, this.sep);
    }
    resolve(path, ...paths) {
        return _this_origin(this).resolve(path, ...paths).replace(/\\/g, this.sep);
    }
    parse(path) {
        let ret = _this_origin(this).parse(path);
        ret.root = ret.root.replace(/\\/g, this.sep);
        ret.dir = ret.dir.replace(/\\/g, this.sep);
        return ret;
    }
    format(pathObject) {
        return _replace_sep(this, _this_origin(this).format(pathObject));
    }
    basename(path, ext) {
        return _this_origin(this).basename(path, ext);
    }
    dirname(path) {
        return _this_origin(this).dirname(path);
    }
    extname(path) {
        return _this_origin(this).extname(path);
    }
    isAbsolute(path) {
        return _this_origin(this).isAbsolute(path);
    }
}
exports.PathWrap = PathWrap;
(function (PathWrap) {
    let __proto__ = {};
    for (let i in Object.getOwnPropertyDescriptors(PathWrap.prototype)) {
        __proto__[i] = PathWrap.prototype[i];
    }
    PathWrap.fn = __proto__;
    PathWrap.fn['fn'] = PathWrap.fn;
    PathWrap.fn.sep = '/';
    PathWrap.prototype.fn = PathWrap.fn;
})(PathWrap = exports.PathWrap || (exports.PathWrap = {}));
function getStatic(who) {
    return who.__proto__.constructor;
}
exports.posix = new PathWrap(_path.posix, 'posix');
exports.win32 = new PathWrap(_path.win32, 'win32');
const _upath = new PathWrap(_path, 'upath');
exports.upath = _upath;
exports.upath.PathWrap = PathWrap;
exports.upath.fn.win32 = exports.win32;
exports.upath.fn.posix = exports.posix;
exports.upath.fn.upath = exports.upath;
exports.upath.fn.default = exports.upath;
exports.fn = PathWrap.fn = exports.upath.fn;
_path.upath = exports.upath;
for (let key of [
    'win32',
    'posix',
    'upath',
]) {
    exports.win32.fn[key] = exports.posix.fn[key] = exports.upath.fn[key] = exports.upath[key];
}
exports.win32.default = exports.posix.default = exports.upath.default = exports.upath;
exports.default = exports.upath;
function _this_origin(who) {
    if (who[type_1.ORIGIN_KEY]) {
        return who[type_1.ORIGIN_KEY];
    }
    else if (who === exports.upath) {
        return _path;
    }
    else if (who === exports.win32) {
        return _path.win32;
    }
    else if (who === exports.posix) {
        return _path.posix;
    }
    throw new TypeError(`this not PathWrap`);
}
function _replace_sep(who, input) {
    return input.replace(/\\/g, who.sep);
}
Object.freeze(exports);
