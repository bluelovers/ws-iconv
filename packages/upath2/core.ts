/**
 * Created by user on 2017/12/9/009.
 */

import _path = require('path');

//export const upath = Object.assign({}, path.win32, {
//	sep: '/',
//});

import types, { IPathNode, IPath, IParse, ORIGIN_KEY, IPathType } from './lib/type';
export { types, IPathNode, IPath, IParse, IPathType }

//const ORIGIN_KEY = Symbol.for('_origin');

export class PathWrap implements IPath
{
	public sep = '/';
	//protected _origin: IPath;
	public name: string;
	public delimiter: string;
	//protected __proto__: IPath;

	public win32: IPath;
	public posix: IPath;
	public upath: PathWrap;
	public default: PathWrap;

	public fn: IPath;

	[ORIGIN_KEY]?: IPathType

	constructor(path, id: string)
	{
		let _static = getStatic(this);

		//this._origin = path;
		this[ORIGIN_KEY] = path;

		this.name = id;

		delete this[id];
		this[id] = this;

		//Object.defineProperty(this, '_origin', { enumerable: false, });
		Object.defineProperty(this, ORIGIN_KEY, { enumerable: false, });

		// @ts-ignore
		this.fn = Object.assign(this.__proto__, path, _static.fn);
	}

	public join<T = string, U = string>(path: T, ...paths: U[]): string
	{
		//console.log(this.name, this.sep);

		return _this_origin(this).join(path, ...paths).replace(/\\/g, this.sep);
	}

	public normalize<T = string>(path: T): string
	{
		return _this_origin(this).normalize(path).replace(/\\/g, this.sep);
	}

	public relative<T = string, U = string>(from: T, to: U): string
	{
		return _this_origin(this).relative(from.toString(), to.toString()).replace(/\\/g, this.sep);
	}

	public resolve<T = string, U = string>(path: T, ...paths: U[]): string
	{
		return _this_origin(this).resolve(path, ...paths).replace(/\\/g, this.sep);
	}

	public parse<T = string>(path: T): IParse
	{
		let ret = _this_origin(this).parse(path);

		ret.root = ret.root.replace(/\\/g, this.sep);
		ret.dir = ret.dir.replace(/\\/g, this.sep);

		return ret;
	}

	public format<T = IParse>(pathObject: T): string
	{
		return _replace_sep(this, _this_origin(this).format(pathObject));
	}

	// ---------

	public basename<T = string, U = string>(path: T, ext?: U): string
	{
		return _this_origin(this).basename(path, ext);
	}

	public dirname<T = string>(path: T): string
	{
		return _this_origin(this).dirname(path);
	}

	public extname<T = string>(path: T): string
	{
		return _this_origin(this).extname(path);
	}

	public isAbsolute<T = string>(path: T): boolean
	{
		return _this_origin(this).isAbsolute(path);
	}
}

export namespace PathWrap
{
	let __proto__ = {};

	// get prototype from class
	for (let i in Object.getOwnPropertyDescriptors(PathWrap.prototype))
	{
		__proto__[i] = PathWrap.prototype[i];
	}

	export let fn = __proto__ as IPath;

	fn['fn'] = fn;
	fn.sep = '/';

	PathWrap.prototype.fn = fn;

	export type IPath = types.IPath;
	export type IPathNode = types.IPathNode;
	export type IParse = types.IParse;
}

function getStatic(who): PathWrap
{
	return who.__proto__.constructor;
}

export const posix = new PathWrap(_path.posix, 'posix') as IPath;
export const win32 = new PathWrap(_path.win32, 'win32') as IPath;
const _upath = new PathWrap(_path, 'upath');

export type IUPath = PathWrap & {
	default: IUPath,
	upath: IUPath,
	PathWrap: typeof PathWrap,
};

export const upath = _upath as IUPath;

upath.PathWrap = PathWrap;

//upath.win32 = win32;
//upath.posix = posix;
//upath.upath = upath;

//PathWrap.fn = upath.fn;

upath.fn.win32 = win32;
upath.fn.posix = posix;
upath.fn.upath = upath;
// @ts-ignore
upath.fn.default = upath;

export const fn = PathWrap.fn = upath.fn;

// @ts-ignore
_path.upath = upath;

for (let key of [
	'win32',
	'posix',
	'upath',
])
{
	win32.fn[key] = posix.fn[key] = upath.fn[key] = upath[key];
}

win32.default = posix.default = upath.default = upath;

// @ts-ignore
//export default upath as PathWrap & IPath & IPathNode;
export default upath;

function _this_origin(who: IPath): IPathNode
{
	if (who[ORIGIN_KEY])
	{
		// @ts-ignore
		return who[ORIGIN_KEY];
	}
	else if (who === upath)
	{
		// @ts-ignore
		return _path;
	}
	else if (who === win32)
	{
		// @ts-ignore
		return _path.win32;
	}
	else if (who === posix)
	{
		// @ts-ignore
		return _path.posix;
	}

	throw new TypeError(`this not PathWrap`);
}

function _replace_sep(who: IPath, input: string): string
{
	return input.replace(/\\/g, who.sep);
}

// @ts-ignore
Object.freeze(exports)
