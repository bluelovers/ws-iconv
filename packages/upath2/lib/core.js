"use strict";
/**
 * Created by user on 2020/6/9.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports._this_origin = void 0;
const type_1 = require("./type");
const core_1 = __importStar(require("../core"));
const path_1 = __importDefault(require("path"));
function _this_origin(who) {
    if (who[type_1.ORIGIN_KEY]) {
        // @ts-ignore
        return who[type_1.ORIGIN_KEY];
    }
    else if (who === core_1.default) {
        // @ts-ignore
        return path_1.default;
    }
    else if (who === core_1.win32) {
        // @ts-ignore
        return path_1.default.win32;
    }
    else if (who === core_1.posix) {
        // @ts-ignore
        return path_1.default.posix;
    }
    throw new TypeError(`this not PathWrap`);
}
exports._this_origin = _this_origin;
//# sourceMappingURL=core.js.map