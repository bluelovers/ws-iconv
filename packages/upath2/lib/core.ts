/**
 * Created by user on 2020/6/9.
 */

import { IPath, IPathNode, ORIGIN_KEY } from './type';
import upath, { win32, posix } from '../core';
import _path from "path";

export function _this_origin(who: IPath): IPathNode
{
	if (who[ORIGIN_KEY])
	{
		// @ts-ignore
		return who[ORIGIN_KEY];
	}
	else if (who === upath)
	{
		// @ts-ignore
		return _path;
	}
	else if (who === win32)
	{
		// @ts-ignore
		return _path.win32;
	}
	else if (who === posix)
	{
		// @ts-ignore
		return _path.posix;
	}

	throw new TypeError(`this not PathWrap`);
}
