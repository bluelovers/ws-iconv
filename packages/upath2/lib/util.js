"use strict";
/**
 * Created by user on 2020/6/9.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getStatic = exports._replace_sep = exports._strip_sep = void 0;
const tslib_1 = require("tslib");
const path_is_network_drive_1 = (0, tslib_1.__importDefault)(require("path-is-network-drive"));
const path_strip_sep_1 = (0, tslib_1.__importDefault)(require("path-strip-sep"));
exports._strip_sep = path_strip_sep_1.default;
function _replace_sep(who, input) {
    let sep = who.sep;
    if (who.name !== 'posix' && (0, path_is_network_drive_1.default)(input)) {
        sep = '\\';
        input = sep + sep + input
            .slice(2)
            .replace(/[/\\]/g, sep);
    }
    else {
        input = input
            .replace(/[/\\]/g, sep);
    }
    //	if (/^\w:[/\\]$/.test(input))
    //	{
    //		return input
    //	}
    return input;
}
exports._replace_sep = _replace_sep;
function getStatic(who) {
    return who.__proto__.constructor;
}
exports.getStatic = getStatic;
//# sourceMappingURL=util.js.map