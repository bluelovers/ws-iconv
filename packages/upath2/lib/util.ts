/**
 * Created by user on 2020/6/9.
 */

import { IPath } from './type';
import { PathWrap } from '../core';
import pathIsNetworkDrive from 'path-is-network-drive';

export function _replace_sep(who: Pick<IPath, 'sep' | 'name'>, input: string): string
{
	let sep = who.sep;

	if (who.name !== 'posix' && pathIsNetworkDrive(input))
	{
		sep = '\\';

		input = sep + sep + input
			.slice(2)
			.replace(/[/\\]/g, sep)
		;
	}
	else
	{
		input = input
			.replace(/[/\\]/g, sep)
		;
	}

	if (/^\w:[/\\]$/.test(input))
	{
		return input
	}

	return input
		.replace(/([^/\\])[/\\]$/, '$1')
		;
}

export function getStatic(who): PathWrap
{
	return who.__proto__.constructor;
}
