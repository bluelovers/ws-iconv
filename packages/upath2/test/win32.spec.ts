import { inspect } from "util";
import upath from "upath2";
import { win32 as path } from '../index';
import testCases, { ITestCaseRow } from './data/testcases';
import { IPath, win32 } from '../core';

describe(`normalize`, () =>
{
	const name = `normalize`;

	const testCases: ITestCaseRow[] = ([
		[
			'//windows\\unix/mixed',
			"\\\\windows\\unix\\mixed",
		],
		[
			'\\windows//unix/mixed',
			"/windows/unix/mixed"
		],
		[
			'\\\\storageserver01\\Active Projects\\ProjectA',
			'\\\\storageserver01\\Active Projects\\ProjectA',
		]
	]);

	let fn = path[name].bind(path);

	_test(fn, name, testCases);

	fn = path[name];

	_test(fn, `(unbind) ${name}`, testCases);

})

function _test(fn, name: string, testCases: ITestCaseRow[])
{
	describe(name, () =>
	{
		testCases
			.forEach(testCase =>
			{
				test(inspect(testCase[0]), () =>
				{
					let actual;
					let expected
					let expected2

					if (typeof testCase[0] === 'string')
					{
						actual = fn(testCase[0])
						expected2 = upath[name]?.(testCase[0])
						expected = testCase[1]

					}
					else
					{
						actual = fn(...testCase[0])
						expected2 = upath[name]?.(...testCase[0])
						expected = testCase[1]
					}

					if (typeof expected !== "undefined")
					{
						expect(actual).toStrictEqual(expected)
					}

					console.dir({
						upath2: actual,
						upath: expected2,
					})

					expect(actual).toMatchSnapshot();
				});
			})
		;

	});
}
