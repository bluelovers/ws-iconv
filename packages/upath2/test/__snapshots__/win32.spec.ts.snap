// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`normalize (unbind) normalize '//windows\\\\unix/mixed' 1`] = `"\\\\\\\\windows\\\\unix\\\\mixed"`;

exports[`normalize (unbind) normalize '\\\\\\\\storageserver01\\\\Active Projects\\\\ProjectA' 1`] = `"\\\\\\\\storageserver01\\\\Active Projects\\\\ProjectA"`;

exports[`normalize (unbind) normalize '\\\\windows//unix/mixed' 1`] = `"/windows/unix/mixed"`;

exports[`normalize normalize '//windows\\\\unix/mixed' 1`] = `"\\\\\\\\windows\\\\unix\\\\mixed"`;

exports[`normalize normalize '\\\\\\\\storageserver01\\\\Active Projects\\\\ProjectA' 1`] = `"\\\\\\\\storageserver01\\\\Active Projects\\\\ProjectA"`;

exports[`normalize normalize '\\\\windows//unix/mixed' 1`] = `"/windows/unix/mixed"`;
